diff -Naur u-boot-1.1.4/net/bootp.c u-boot-1.1.4-bugfix/net/bootp.c
--- u-boot-1.1.4/net/bootp.c	2005-12-16 11:39:27.000000000 -0500
+++ u-boot-1.1.4-bugfix/net/bootp.c	2006-04-19 14:45:17.871390684 -0400
@@ -120,9 +120,15 @@
 	IPaddr_t tmp_ip;
 
 	NetCopyIP(&NetOurIP, &bp->bp_yiaddr);
+
+	/* This is the DHCP/BOOTP Server Address. It may not be our boot server */
+	/* So copy only if we have not already set our boot server */
+	if (NetServerIP == 0)
+	{
 	NetCopyIP(&tmp_ip, &bp->bp_siaddr);
 	if (tmp_ip != 0)
 		NetCopyIP(&NetServerIP, &bp->bp_siaddr);
+	}
 	memcpy (NetServerEther, ((Ethernet_t *)NetRxPkt)->et_src, 6);
 	if (strlen(bp->bp_file) > 0)
 		copy_filename (BootFile, bp->bp_file, sizeof(BootFile));
