diff -Naur u-boot-1.1.4.org/net/net.c u-boot-1.1.4/net/net.c
--- u-boot-1.1.4.org/net/net.c	2006-07-10 15:50:15.735071463 -0400
+++ u-boot-1.1.4/net/net.c	2006-07-10 11:43:31.000000000 -0400
@@ -168,6 +168,11 @@
 int nc_input_packet(uchar *pkt, unsigned dest, unsigned src, unsigned len);
 #endif
 
+
+// Added for AMI Extension - R2C
+extern void R2C_Initiate(char * Ether);
+extern int do_reset (cmd_tbl_t *cmdtp, int flag, int argc, char *argv[]);
+
 volatile uchar	PktBuf[(PKTBUFSRX+1) * PKTSIZE_ALIGN + PKTALIGN];
 
 volatile uchar *NetRxPackets[PKTBUFSRX]; /* Receive packets			*/
@@ -448,6 +453,10 @@
 			SntpStart();
 			break;
 #endif
+		/* AMI Extension - R2C */
+		case R2C:
+			R2C_Initiate(NetOurEther);
+			break;
 		default:
 			break;
 		}
@@ -491,9 +500,17 @@
 		 *	Abort if ctrl-c was pressed.
 		 */
 		if (ctrlc()) {
-			eth_halt();
-			puts ("\nAbort\n");
-			return (-1);
+			if (protocol != R2C)
+			{
+				eth_halt();
+				puts ("\nAbort\n");
+				return (-1);
+			}
+			else		/* AMI Extension - R2C */
+			{
+				printf("\nR2C should not be aborted. Resetting ...\n");
+				do_reset(NULL,0,0,NULL);
+			}
 		}
 
 		ArpTimeoutCheck();
