diff -Naur u-boot-1.1.4.org/include/net.h u-boot-1.1.4/include/net.h
--- u-boot-1.1.4.org/include/net.h	2006-07-10 15:50:17.771186425 -0400
+++ u-boot-1.1.4/include/net.h	2006-07-10 11:38:01.000000000 -0400
@@ -336,7 +336,9 @@
 extern int		NetRestartWrap;		/* Tried all network devices	*/
 #endif
 
-typedef enum { BOOTP, RARP, ARP, TFTP, DHCP, PING, DNS, NFS, CDP, NETCONS, SNTP } proto_t;
+//typedef enum { BOOTP, RARP, ARP, TFTP, DHCP, PING, DNS, NFS, CDP, NETCONS, SNTP } proto_t;
+//AMI Extension - R2C Protocol
+typedef enum { BOOTP, RARP, ARP, TFTP, DHCP, PING, DNS, NFS, CDP, NETCONS, SNTP, R2C } proto_t;
 
 /* from net/net.c */
 extern char	BootFile[128];			/* Boot File name		*/
