diff -Naur u-boot-1.1.4/config.mk u-boot-1.1.4-bugfix/config.mk
--- u-boot-1.1.4/config.mk	2005-12-16 11:39:27.000000000 -0500
+++ u-boot-1.1.4-bugfix/config.mk	2006-04-19 14:48:02.800499375 -0400
@@ -181,10 +181,13 @@
 #########################################################################
 
 %.s:	%.S
-	$(CPP) $(AFLAGS) -o $@ $(CURDIR)/$<
+	@echo "Preprocessing $< ..."
+	@$(CPP) $(AFLAGS) -o $@ $(CURDIR)/$<
 %.o:	%.S
-	$(CC) $(AFLAGS) -c -o $@ $(CURDIR)/$<
+	@echo "Assempling $< ..."
+	@$(CC) $(AFLAGS) -c -o $@ $(CURDIR)/$<
 %.o:	%.c
-	$(CC) $(CFLAGS) -c -o $@ $<
+	@echo "Compiling $< ..."
+	@$(CC) $(CFLAGS) -c -o $@ $<
 
 #########################################################################
