diff -Naur u-boot-1.1.4/Makefile u-boot-1.1.4-bugfix/Makefile
--- u-boot-1.1.4/Makefile	2005-12-16 11:39:27.000000000 -0500
+++ u-boot-1.1.4-bugfix/Makefile	2006-04-19 14:45:55.577461091 -0400
@@ -1826,6 +1826,7 @@
 	      examples/eepro100_eeprom examples/sched \
 	      examples/mem_to_mem_idma2intr examples/82559_eeprom \
 	      examples/test_burst
+	rm -f diag/*.elf
 	rm -f tools/img2srec tools/mkimage tools/envcrc tools/gen_eth_addr
 	rm -f tools/mpc86x_clk tools/ncb
 	rm -f tools/easylogo/easylogo tools/bmp_logo
