diff -Naur u-boot-1.1.4.org/common/main.c u-boot-1.1.4/common/main.c
--- u-boot-1.1.4.org/common/main.c	2006-07-10 15:50:15.679068301 -0400
+++ u-boot-1.1.4/common/main.c	2006-07-10 15:51:37.235672802 -0400
@@ -261,13 +261,15 @@
 		/* delay 100 * 10ms */
 		for (i=0; !abort && i<100; ++i) {
 			if (tstc()) {	/* we got a key press	*/
-				abort  = 1;	/* don't auto boot	*/
-				bootdelay = 0;	/* no more delay	*/
 # ifdef CONFIG_MENUKEY
 				menukey = getc();
+				if (menukey != CONFIG_MENUKEY) 		// AMI Addition
+					continue;
 # else
 				(void) getc();  /* consume input	*/
 # endif
+				abort  = 1;	/* don't auto boot	*/
+				bootdelay = 0;	/* no more delay	*/
 				break;
 			}
 			udelay (10000);
