--- uboot/cpu/arm926ejs/start.S	2007-04-20 15:43:33.000000000 -0400
+++ uboot.new/cpu/arm926ejs/start.S	2007-04-20 15:46:15.000000000 -0400
@@ -286,13 +286,13 @@
 	.macro	irq_save_user_regs
 	sub	sp, sp, #S_FRAME_SIZE
 	stmia	sp, {r0 - r12}			@ Calling r0-r12
-	@ !!!! R8 NEEDS to be saved !!!! a reserved stack spot would be good.
-	add	r8, sp, #S_PC
-	stmdb	r8, {sp, lr}^		@ Calling SP, LR
-	str	lr, [r8, #0]		@ Save calling PC
+
+	add	r7, sp, #S_PC
+	stmdb	r7, {sp, lr}^		@ Calling SP, LR
+	str	lr, [r7, #0]		@ Save calling PC
 	mrs	r6, spsr
-	str	r6, [r8, #4]		@ Save CPSR
-	str	r0, [r8, #8]		@ Save OLD_R0
+	str	r6, [r7, #4]		@ Save CPSR
+	str	r0, [r7, #8]		@ Save OLD_R0
 	mov	r0, sp
 	.endm
 
